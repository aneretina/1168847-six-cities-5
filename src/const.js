export const ZOOM = 12;

export const TO_PERCENT = 20;

export const CITIES = [
  `Amsterdam`,
  `Paris`,
  `Cologne`,
  `Hamburg`,
  `Dusseldorf`,
  `Brussels`
];


export const Icon = {
  URL: `/img/pin.svg`,
  ACTIVE_URL: `img/pin-active.svg`,
  SIZE: [30, 30],
};

export const ID_MAP_CONTAINER = `map`;

export const SortOptions = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`
};

export const AuthorizationStatus = {
  NOT_AUTHORIZED: `NOT_AUTHORIZED`,
  AUTHORIZED: `AUTHORIZED`,
};

export const FavoriteBtnType = {
  PROPERTY: `property`,
  CARD: `place-card`
};

export const FavoriteBtnSize = {
  [FavoriteBtnType.PROPERTY]: {width: 31, height: 33},
  [FavoriteBtnType.CARD]: {width: 18, height: 19},
};

