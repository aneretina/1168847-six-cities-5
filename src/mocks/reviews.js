export const reviewsMock = [
  {
    id: 64,
    date: new Date(`Tue Oct 27 2020 12:29:21 GMT+0300`),
    text: `We would like to thank all the staff, who were very friendly during our stay.`,
    rating: 4,
    avatar: `img/avatar-max.jpg`,
    userName: `Tom`,
    userId: 4,
    userIsPro: true
  },
  {
    id: 1,
    date: new Date(`Tue Oct 27 2020 12:29:21 GMT+0300`),
    text: `The hotel had everything we needed and it was in an excellent location, being in a lovely area with lots of places to enjoy ourselves.`,
    rating: 5,
    avatar: `img/avatar-max.jpg`,
    userName: `Peter`,
    userId: 1,
    userIsPro: true
  },
  {
    id: 18,
    date: new Date(`Tue Oct 27 2020 12:29:21 GMT+0300`),
    text: `Iâ€™ve been many times and I will come again and again.`,
    rating: 5,
    avatar: `img/avatar-max.jpg`,
    userName: `Michael`,
    userId: 1,
    userIsPro: true
  },
];

