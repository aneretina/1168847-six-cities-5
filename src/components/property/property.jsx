import React, {PureComponent} from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import OfferMap from "../offer-map/offer-map";
import OffersList from "../offers-list/offers-list";
import ReviewForm from "../review-form/review-form";
import {getCurrentCityOffers, getCurrentCity, getAuthorizationStatus, getNearOffers, getReviews} from "../../store/selectors/selectors";
import Header from "../ header/header";
import {TO_PERCENT, FavoriteBtnType, AuthorizationStatus} from "../../const";
import {changeFavoriteStatus, fetchNearOffersList, fetchReviewsList} from "../../store/api-actions";
import offerProp from "../offer-card/offer.prop";
import ReviewsList from "../reviews-list/reviews-list";
import reviewProp from "../review/review.prop";
import FavoriteButton from "../favorite-button/favorite-button";
import browserHistory from "../../browser-history";


class Property extends PureComponent {
  componentDidMount() {
    this.props.loadNearOffersAction(this.props.id);
    this.props.loadReviewsAction(this.props.id);
  }

  render() {
    const {offers, reviews, id, activeCity, authorizationStatus, changeFavoriteStatusAction, nearOffers} = this.props;
    const offer = offers.find((offerCurrent) => offerCurrent.id === Number(id));

    const picturesForShow = offer.pictures.slice(0, 6);

    const onFavoriteButtonClick = () => {
      if (this.props.authorizationStatus === AuthorizationStatus.NOT_AUTHORIZED) {
        browserHistory.push(`/login`);
      }

      changeFavoriteStatusAction(offer.id, !offer.isFavorite ? 1 : 0);
    };

    return (
      <div className="page">
        <Header />
        <main className="page__main page__main--property">
          <section className="property">
            <div className="property__gallery-container container">
              <div className="property__gallery">
                {picturesForShow.map((img, i) => (
                  <div key={i} className="property__image-wrapper">
                    <img className="property__image" src={img} alt={offer.title}/>
                  </div>
                ))}
              </div>
            </div>
            <div className="property__container container">
              <div className="property__wrapper">
                <div className="property__mark">
                  {offer.isPremium ?
                    <div className="place-card__mark">
                      <span>Premium</span>
                    </div>
                    : ``}
                </div>
                <div className="property__name-wrapper">
                  <h1 className="property__name">
                    {offer.title}
                  </h1>
                  <FavoriteButton
                    type={FavoriteBtnType.PROPERTY}
                    isFavorite={offer.isFavorite}
                    onFavoriteButtonClick={onFavoriteButtonClick}
                  />
                </div>
                <div className="property__rating rating">
                  <div className="property__stars rating__stars">
                    <span style={{width: offer.rating * TO_PERCENT + `%`}}></span>
                    <span className="visually-hidden">Rating</span>
                  </div>
                  <span className="property__rating-value rating__value">{offer.rating}</span>
                </div>
                <ul className="property__features">
                  <li className="property__feature property__feature--entire">
                    {offer.type}
                  </li>
                  <li className="property__feature property__feature--bedrooms">
                    {offer.bedroomsCount} Bedrooms
                  </li>
                  <li className="property__feature property__feature--adults">
                    {offer.guestsLimit} adults
                  </li>
                </ul>
                <div className="property__price">
                  <b className="property__price-value">&euro;{offer.price}</b>
                  <span className="property__price-text">&nbsp;night</span>
                </div>
                <div className="property__inside">
                  <h2 className="property__inside-title">What&apos;s inside</h2>
                  <ul className="property__inside-list">
                    {offer.goods.map((feature, index) => {
                      return (
                        <li key={index} className="property__inside-item">
                          {feature}
                        </li>);
                    })}
                  </ul>
                </div>
                <div className="property__host">
                  <h2 className="property__host-title">Meet the host</h2>
                  <div className="property__host-user user">
                    <div className={`property__avatar-wrapper ${offer.host.isPro ? `property__avatar-wrapper--pro` : ``} user__avatar-wrapper`}>
                      <img className="property__avatar user__avatar" src={offer.host.avatar} width="74" height="74" alt="Host avatar"/>
                    </div>
                    <span className="property__user-name">
                      {offer.host.name}
                    </span>
                  </div>
                  <div className="property__description">
                    {offer.description}
                  </div>
                </div>
                <section className="property__reviews reviews">
                  <ReviewsList reviews={reviews} />
                  {authorizationStatus === AuthorizationStatus.AUTHORIZED &&
                  <ReviewForm id={id} />
                  }
                </section>
              </div>
            </div>
            {nearOffers.length > 0 &&
            <OfferMap offers={nearOffers}
              activeCity={activeCity}
              cityCoords={[nearOffers[0].city.location.latitude, nearOffers[0].city.location.longitude]}
              zoom={nearOffers[0].city.location.zoom}
              mainOffer={offer}
              className={`property__map`} />
            }
          </section>
          <div className="container">
            <section className="near-places places">
              {nearOffers.length > 0 &&
              <>
                <h2 className="near-places__title">Other places in the neighbourhood</h2>

                <OffersList offers={nearOffers}
                  authorizationStatus={authorizationStatus}
                  changeFavoriteStatusAction={changeFavoriteStatusAction}
                  className={`near-places__list`} />
              </>
              }
            </section>

          </div>
        </main>
      </div>
    );
  }
}

Property.propTypes = {
  nearOffers: PropTypes.array.isRequired,
  loadNearOffersAction: PropTypes.func.isRequired,
  loadReviewsAction: PropTypes.func.isRequired,
  activeCity: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  offers: PropTypes.arrayOf(offerProp),
  authorizationStatus: PropTypes.string.isRequired,
  changeFavoriteStatusAction: PropTypes.func.isRequired,
  reviews: PropTypes.arrayOf(reviewProp),
};

const mapStateToProps = (state) => ({
  offers: getCurrentCityOffers(state),
  activeCity: getCurrentCity(state),
  authorizationStatus: getAuthorizationStatus(state),
  nearOffers: getNearOffers(state),
  reviews: getReviews(state),
});

const mapDispatchToProps = ((dispatch) => ({
  changeFavoriteStatusAction(id, num) {
    dispatch(changeFavoriteStatus(id, num));
  },

  loadNearOffersAction(id) {
    dispatch(fetchNearOffersList(id));
  },

  loadReviewsAction(id) {
    dispatch(fetchReviewsList(id));
  },
}));

export {Property};
export default connect(mapStateToProps, mapDispatchToProps)(Property);

