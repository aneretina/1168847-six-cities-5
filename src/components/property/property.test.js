import React from 'react';
import renderer from 'react-test-renderer';
import {AuthorizationStatus} from '../../const';
import {reviewsMock} from '../../mocks/reviews';
import {Property} from './property';
import {BrowserRouter} from 'react-router-dom';
import {createStore} from 'redux';
import rootReducer from '../../store/reducers/root/root-reducer';
import {Provider} from 'react-redux';
import {offersMock} from '../../mocks/offers';

jest.mock(`../offer-map/offer-map`, () => `OfferMap`);

const store = createStore(rootReducer);

it(`Should Property render properly`, () => {
  const noop = () => {};
  const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Property
                nearOffers={offersMock}
                loadNearOffersAction={noop}
                loadReviewsAction={noop}
                activeCity={`Paris`}
                id={1}
                offers={offersMock}
                authorizationStatus={AuthorizationStatus.AUTHORIZED}
                changeFavoriteStatusAction={noop}
                reviews={reviewsMock}
              />
            </BrowserRouter>
          </Provider>)
      .toJSON();

  expect(tree).toMatchSnapshot();
});
