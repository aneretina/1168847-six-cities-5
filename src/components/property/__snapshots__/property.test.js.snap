// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Property render properly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment in Paris"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment in Paris"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment in Paris"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment in Paris"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment in Paris"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment in Paris"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious apartment in Paris
            </h1>
            <button
              className="property__bookmark-button property__bookmark-button--active button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height={33}
                width={31}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              1
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              2
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              80
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wi-Fi
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Parking
              </li>
              <li
                className="property__inside-item"
              >
                Flowers
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                TV
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-maggy.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Maggy
              </span>
            </div>
            <div
              className="property__description"
            >
              Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                3
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Tom
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    We would like to thank all the staff, who were very friendly during our stay.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-10-27"
                  >
                    October 2020
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Peter
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    The hotel had everything we needed and it was in an excellent location, being in a lovely area with lots of places to enjoy ourselves.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-10-27"
                  >
                    October 2020
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Michael
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    I’ve been many times and I will come again and again.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-10-27"
                  >
                    October 2020
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form"
              method="post"
              onSubmit={[Function]}
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="5-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="5"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="4-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="4"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="3-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="3"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="2-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="2"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="1-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="1"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-stars"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                disabled={false}
                id="review"
                name="review"
                onChange={[Function]}
                placeholder="Tell how was your stay, what you like and what can be improved"
                value=""
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
              <p
                className="error"
                style={
                  Object {
                    "color": "red",
                  }
                }
              />
            </form>
          </section>
        </div>
      </div>
      <OfferMap
        activeCity="Paris"
        cityCoords={
          Array [
            52.38333,
            4.9,
          ]
        }
        className="property__map"
        mainOffer={
          Object {
            "bedroomsCount": 1,
            "city": Object {
              "location": Object {
                "latitude": 48.864716,
                "longitude": 2.349014,
              },
              "name": "Paris",
            },
            "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
            "goods": Array [
              "Wi-Fi",
              "Heating",
              "Kitchen",
              "Parking",
              "Flowers",
              "Dishwasher",
              "Towels",
              "TV",
            ],
            "guestsLimit": 2,
            "host": Object {
              "avatar": "img/avatar-maggy.jpg",
              "isPro": true,
              "name": "Maggy",
            },
            "id": 1,
            "isFavorite": true,
            "isPremium": true,
            "location": Object {
              "latitude": 48.87456,
              "longitude": 2.413,
            },
            "pictures": Array [
              "img/apartment-02.jpg",
              "img/apartment-02.jpg",
              "img/apartment-02.jpg",
              "img/apartment-02.jpg",
              "img/apartment-02.jpg",
              "img/apartment-02.jpg",
            ],
            "previewImage": "img/apartment-02.jpg",
            "price": 80,
            "rating": 4,
            "title": "Beautiful & luxurious apartment in Paris",
            "type": "Apartment",
          }
        }
        offers={
          Array [
            Object {
              "bedroomsCount": 2,
              "city": Object {
                "location": Object {
                  "latitude": 52.38333,
                  "longitude": 4.9,
                },
                "name": "Amsterdam",
              },
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "goods": Array [
                "Wi-Fi",
                "Heating",
                "Kitchen",
                "Parking",
                "Flowers",
                "Dishwasher",
                "Towels",
                "TV",
              ],
              "guestsLimit": 2,
              "host": Object {
                "avatar": "img/avatar-max.jpg",
                "isPro": false,
                "name": "Max",
              },
              "id": 15,
              "isFavorite": true,
              "isPremium": true,
              "location": Object {
                "latitude": 52.3909553943508,
                "longitude": 4.85309666406198,
              },
              "pictures": Array [
                "img/apartment-01.jpg",
                "img/apartment-01.jpg",
                "img/apartment-01.jpg",
                "img/apartment-01.jpg",
                "img/apartment-01.jpg",
                "img/apartment-01.jpg",
              ],
              "previewImage": "img/apartment-01.jpg",
              "price": 180,
              "rating": 3,
              "title": "Wood and stone place",
              "type": "room",
            },
            Object {
              "bedroomsCount": 1,
              "city": Object {
                "location": Object {
                  "latitude": 48.864716,
                  "longitude": 2.349014,
                },
                "name": "Paris",
              },
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "goods": Array [
                "Wi-Fi",
                "Heating",
                "Kitchen",
                "Parking",
                "Flowers",
                "Dishwasher",
                "Towels",
                "TV",
              ],
              "guestsLimit": 2,
              "host": Object {
                "avatar": "img/avatar-maggy.jpg",
                "isPro": true,
                "name": "Maggy",
              },
              "id": 1,
              "isFavorite": true,
              "isPremium": true,
              "location": Object {
                "latitude": 48.87456,
                "longitude": 2.413,
              },
              "pictures": Array [
                "img/apartment-02.jpg",
                "img/apartment-02.jpg",
                "img/apartment-02.jpg",
                "img/apartment-02.jpg",
                "img/apartment-02.jpg",
                "img/apartment-02.jpg",
              ],
              "previewImage": "img/apartment-02.jpg",
              "price": 80,
              "rating": 4,
              "title": "Beautiful & luxurious apartment in Paris",
              "type": "Apartment",
            },
            Object {
              "bedroomsCount": 4,
              "city": Object {
                "location": Object {
                  "latitude": 50.935173,
                  "longitude": 6.953101,
                },
                "name": "Cologne",
              },
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "goods": Array [
                "Wi-Fi",
                "Heating",
                "Kitchen",
                "Parking",
                "Flowers",
                "Dishwasher",
                "Towels",
                "TV",
              ],
              "guestsLimit": 5,
              "host": Object {
                "avatar": "img/avatar-bill.jpg",
                "isPro": true,
                "name": "Bill",
              },
              "id": 9,
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 50.9682,
                "longitude": 6.874,
              },
              "pictures": Array [
                "img/apartment-03.jpg",
                "img/apartment-03.jpg",
                "img/apartment-03.jpg",
                "img/apartment-03.jpg",
                "img/apartment-03.jpg",
                "img/apartment-03.jpg",
              ],
              "previewImage": "img/apartment-03.jpg",
              "price": 220,
              "rating": 1,
              "title": "Beautiful & luxurious apartment in Cologne",
              "type": "Apartment",
            },
          ]
        }
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="places__list near-places__list"
        >
          <article
            className="cities__place-card place-card"
            id={15}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/15"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-01.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    180
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/15"
                  onClick={[Function]}
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                room
              </p>
            </div>
          </article>
          <article
            className="cities__place-card place-card"
            id={1}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/1"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-02.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    80
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/1"
                  onClick={[Function]}
                >
                  Beautiful & luxurious apartment in Paris
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
          <article
            className="cities__place-card place-card"
            id={9}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/9"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-03.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    220
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "20%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/9"
                  onClick={[Function]}
                >
                  Beautiful & luxurious apartment in Cologne
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
